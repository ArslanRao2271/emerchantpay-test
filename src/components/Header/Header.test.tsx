import { render, screen } from "@testing-library/react";
import { Header } from ".";
import { MemoryRouter } from "react-router-dom";

const getWrapperComponent = (comp: JSX.Element) => {
  return render(comp, { wrapper: MemoryRouter });
};

describe("Header test cases", () => {
  test("Should match previous snapshot", () => {
    const { asFragment } = getWrapperComponent(<Header />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Should render Header text", () => {
    getWrapperComponent(<Header />);

    expect(screen.queryByText(/E-MerchantPay/)).toBeVisible();
  });
});
