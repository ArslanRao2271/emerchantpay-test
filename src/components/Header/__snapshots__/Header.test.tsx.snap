// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test cases Should match previous snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-qcqlej"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1m7hp5s-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1juivf6-MuiTypography-root"
        >
          E-MerchantPay
        </h6>
        <div
          class="MuiBox-root css-0"
        >
          <button
            aria-haspopup="true"
            aria-label="menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeLarge css-ancrnh-MuiButtonBase-root-MuiIconButton-root"
            id="basic-button"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
